gradle/scripts/BranchName.gradle

```diff
@@ -0,0 +1,23 @@
+task checkBranchName {
+  doLast {
+    def branchName = System.getenv('GITHUB_REF_NAME')
+    def eventName = System.getenv('GITHUB_EVENT_NAME')
+    def headRef = System.getenv('GITHUB_HEAD_REF')
+
+    if (eventName == 'pull_request') {
+      branchName = headRef
+    }
+
+    if (branchName.startsWith('refs/heads/')) {
+      branchName = branchName.replaceFirst('refs/heads/', '')
+    }
+
+    def pattern = /^[a-z]{2}\/#([0-9]+)(-.*)?$/
```

Comment:
The capture groups don't hurt much, but are also not necessary I think.

This might work too: `^[a-z]{2}\/#[0-9]+(?:-.+)?$`